bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java

```diff
@@ -935,21 +917,6 @@ public void run() {
             }
         }
 
-        public void pongReceived() {
-            pong = true;
-            connectionConfirmed();
-        }
-
-        private void updateDutyCycleRatio() throws IOException {
-            ListBidcosInterfacesParser parser = getRpcClient(getDefaultInterface())
-                    .listBidcosInterfaces(getDefaultInterface());
-            Integer dutyCycleRatio = parser.getDutyCycleRatio();
-
-            if (dutyCycleRatio != null) {
-                gatewayAdapter.onDutyCycleRatioUpdate(dutyCycleRatio);
-            }
-        }
-
```

Comment:
What does removing these fix?

Reply:
Before the change the binding sent every 30s a ping request to the gateway and it also requested the current duty cycle value. In the case, the gateway was disconnected or unreachable for a time the combination of both requests causes problems (this at least partially is a problem of the Homematic gateway) and the reconnect only partially worked.

Also, the ping request was not supported by older CCU types and the request for the duty cycle was already used as 
a bypass.

In order to find out whether the gateway is connected or not the request for the duty cycle value is sufficient. Therefore I have removed the ping request. This makes method pongReceived() unnecessary. 

The separate method updateDutyCycleRatio() is no longer necessary because the same logic was already existing in the now refactored run() method. I only had to remove an if statement.

Reply:
sounds good to me.